File: leetcode2/0897. Increasing Order Search Tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution(object):
    def increasingBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        if not root:
            return None
        if not root.left and not root.right:
            return root
        elif not root.left and root.right:
            root.right=self.increasingBST(root.right)
            return root
        else:# root.left:
            newroot=self.increasingBST(root.left)
            cur_root=newroot
            while cur_root.right:
                cur_root=cur_root.right
            root.left=None
            cur_root.right=self.increasingBST(root)
            return newroot

File: archives/C#/0897. Increasing Order Search Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public TreeNode IncreasingBST(TreeNode root) {
        if(root==null){
            return null;
        }
        TreeNode newroot=new TreeNode(0);
        if(root.left!=null){
            newroot.right=IncreasingBST(root.left);
        }
        TreeNode curRoot=newroot;
        while(curRoot.right!=null){
            curRoot=curRoot.right;
        }
        curRoot.right=root;
        curRoot=curRoot.right;
        curRoot.left=null;
        curRoot.right=IncreasingBST(root.right);
        
        return newroot.right;
            
    }
}
